code/model/SiteTree.php

```diff
@@ -2217,6 +2217,7 @@ public function fieldLabels($includerelations = true) {
 	 */
 	public function getCMSActions() {
 		$existsOnLive = $this->getExistsOnLive();
+		$numChildren = count($this->getDescendantIDList());
```

Comment:
My concern with this is that selecting all ids just to get a count is a heavily intensive process and could slow down some sites significantly. Can we use `numChildren` instead, and have a less precise (but faster) behaviour?

In master (https://github.com/silverstripe/silverstripe-cms/pull/1774) I think there is a similar problem, which I might go back to and address.